runtime/darwinia/src/pallets/polkadot_xcm.rs

```diff
@@ -215,3 +221,63 @@ impl cumulus_pallet_xcm::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type XcmExecutor = xcm_executor::XcmExecutor<XcmExecutorConfig>;
 }
+
+pub struct EthereumXcmEnsureProxy;
+impl xcm_primitives::EnsureProxy<AccountId> for EthereumXcmEnsureProxy {
+	fn ensure_ok(delegator: AccountId, delegatee: AccountId) -> Result<(), &'static str> {
+		// The EVM implicitely contains an Any proxy, so we only allow for "Any" proxies
+		let def: pallet_proxy::ProxyDefinition<AccountId, pallets::proxy::ProxyType, BlockNumber> =
+			pallet_proxy::Pallet::<Runtime>::find_proxy(
+				&delegator,
+				&delegatee,
+				Some(pallets::proxy::ProxyType::Any),
+			)
+			.map_err(|_| "proxy error: expected `ProxyType::Any`")?;
+		// We only allow to use it for delay zero proxies, as the call will immediatly be executed
+		frame_support::ensure!(def.delay.is_zero(), "proxy delay is Non-zero`");
+		Ok(())
+	}
+}
+
+impl pallet_ethereum_xcm::Config for Runtime {
+	type ControllerOrigin = frame_system::EnsureRoot<AccountId>;
+	type EnsureProxy = EthereumXcmEnsureProxy;
+	type InvalidEvmTransactionError = pallet_ethereum::InvalidTransactionWrapper;
+	type ReservedXcmpWeight =
+		<Runtime as cumulus_pallet_parachain_system::Config>::ReservedXcmpWeight;
+	type ValidatedTransaction = pallet_ethereum::ValidatedTransaction<Self>;
+	type XcmEthereumOrigin = pallet_ethereum_xcm::EnsureXcmEthereumTransaction;
+}
+
+pub struct DarwiniaCall;
```

Comment:
`DarwiniaCall`? This looks like a filter to me.

Reply:
Yes, it wraps Origin for ethereum_xcm transactions, making it equal to [`pallet_ethereum_xcm::Config::XcmEthereumOrigin`](https://github.com/darwinia-network/darwinia/blob/guantong/xcm-evm/runtime/darwinia/src/pallets/polkadot_xcm.rs#L249), and will be checked when [executing ethereum_xcm transactions](https://github.com/PureStake/moonbeam/blob/master/pallets/ethereum-xcm/src/lib.rs#L152)

Reply:
Might need a better name for this.

Reply:
`XcmCallFilter` ?